language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java

```diff
@@ -153,4 +183,149 @@ function testF() {
                 && (returnTypeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR
                 || returnTypeSymbol.get().typeKind() == TypeDescKind.NONE));
     }
+
+    @Override
+    public CodeAction resolve(ResolvableCodeAction codeAction, CodeActionResolveContext resolveContext) {
+        String uri = codeAction.getData().getFileUri();
+        Optional<Path> filePath = PathUtil.getPathFromURI(uri);
+        if (filePath.isEmpty()) {
+            throw new UserErrorException("Invalid parameters received for the create function command!");
+        }
+
+        SyntaxTree syntaxTree = resolveContext.workspace().syntaxTree(filePath.get()).orElseThrow();
+        NonTerminalNode cursorNode = CommonUtil.findNode(codeAction.getData().getRange(), syntaxTree);
+        if (cursorNode == null) {
+            return null;
```

Comment:
We cannot return `null` here (or in any of the following cases) because as per the protocol, we should return the `codeAction` or an error. Here, we can throw a UserErrorException then with a proper error message like `failed to resolve code action`.